### PR TITLE
[opentelemetry-collector] add security best practices documentation

### PR DESCRIPTION
**Pull request by povilasv**
Wrote a little doc explaining the situation in https://github.com/open-telemetry/opentelemetry-helm-charts/pull/497. Let me know what you thing :)

This is how it looks rendered https://github.com/open-telemetry/opentelemetry-helm-charts/blob/b39998c8be1aa405d861e1ea29b72ee6c99c2419/charts/opentelemetry-collector/README.md

### CODE REVIEW

File: charts/opentelemetry-collector/tmp-values.yaml [hunk 1]
```diff
@@ -0,0 +1,366 @@
+# Valid values are "daemonset", "deployment", and "statefulset".
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/tmp-values.yaml:1]: Holdover from another PR?

File: charts/opentelemetry-collector/README.md [hunk 2]
```diff
@@ -25,6 +25,16 @@ helm install my-opentelemetry-collector open-telemetry/opentelemetry-collector
 
 See [UPGRADING.md](UPGRADING.md).
 
+## Security Considerations
+
+OpenTelemetry Collector recommends to bind receivers' servers to addresses that limit connections to authorized users. This is typically not needed in containerized environments, although Open Telemetry Collector logs the following:
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/README.md:30]:
```suggestion
OpenTelemetry Collector recommends to bind receivers' servers to addresses that limit connections to authorized users. This is typically not needed in containerized environments, although the Open Telemetry Collector logs the following:
```

File: charts/opentelemetry-collector/README.md [hunk 3]
```diff
@@ -25,6 +25,16 @@ helm install my-opentelemetry-collector open-telemetry/opentelemetry-collector
...
+```
+Using the 0.0.0.0 address exposes this server to every network interface, which may facilitate Denial of Service attacks
+```
+More info is available in the [Security Best Practices docummentation](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/security-best-practices.md#safeguards-against-denial-of-service-attacks)
+
+Some care must be taken when using `hostNetwork: true`, as then OpenTelemetry Collector will listen on all the address in the host network namespace.
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/README.md:36]:
```suggestion
Some care must be taken when using `hostNetwork: true`, as then OpenTelemetry Collector will listen on all the addresses in the host network namespace.
```

**Review by TylerHelmuth — COMMENTED** [review]
Thanks for adding this

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
LGTM. Thanks for adding this

**Pull request merged**